src/libs/elektra/cmount.c

```diff
@@ -0,0 +1,2919 @@
+			{
+				printf("%s%c", keyName (cur), delim);
+			}
+		}
+
+		if (!clSecond)
+		{
+			printf ("%s%c", keyName (cur), delim);
+		}
+	}
+
+	ksDel (mtab);
+}
+
+
+bool isValidMountPoint (Key * mountPoint, KeySet * mountConf)
+{
+	KeySet * backendInfos = getBackendInfo (mountConf);
+	KeySet * alreadyUsedMountpoints = ksNew (0, KS_END);
+
+	/* handle cascading mountpoints (multiple namespaces) */
+	for (elektraCursor it = 0; it < ksGetSize (backendInfos); ++it)
+	{
+		Key * cur = ksAtCursor (backendInfos, it);
+		if (cur)
+		{
+			const char * curKeyName = keyName (cur);
+			if (keyGetNameSize (cur) == 2 && *curKeyName == '/')
+			{
+				Key * curNamespaceKey = keyNew ("spec:/", KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				curNamespaceKey = keyNew ("dir:/", KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				curNamespaceKey = keyNew ("user:/", KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				curNamespaceKey = keyNew ("system:/", KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+			}
+			else if (keyGetNameSize (cur) > 1 && *curKeyName == '/')
+			{
+				char * tmpStr = elektraMalloc (strlen("system:") + keyGetNameSize(cur)); /* includes \0 */
+
+				strcpy(tmpStr, "dir:");
+				strcat (tmpStr, curKeyName);
+				Key * curNamespaceKey = keyNew (tmpStr, KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				strcpy(tmpStr, "user:");
+				strcat (tmpStr, curKeyName);
+				curNamespaceKey = keyNew (tmpStr, KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				strcpy(tmpStr, "system:");
+				strcat (tmpStr, curKeyName);
+				curNamespaceKey = keyNew (tmpStr, KEY_END);
+				ksAppendKey (alreadyUsedMountpoints, curNamespaceKey);
+
+				elektraFree (tmpStr);
+			}
+
+			/* always add current key itself, too */
+			ksAppendKey (alreadyUsedMountpoints, cur);
+		}
+		else
+		{
+			/* TODO: handle error */
+		}
+	}
+
+	/* STEP 0: CHeck for null key */
+	if (!mountPoint)
+	{
+		// TODO: set error key (was exception in c++ code)
+		fprintf (stderr, "Null mountpoint not allowed!\n");
+		ksDel (backendInfos);
+		ksDel (alreadyUsedMountpoints);
+		return false;
+	}
+
+	/* STEP 1: Check for empty name */
+	if (keyGetNameSize (mountPoint) < 2) /* \0 is counted --> len >= 2 for non-empty string */
+	{
+		// TODO: set error key (was exception in c++ code)
+		fprintf (stderr, "Empty mountpoint not allowed!\n");
+		ksDel (backendInfos);
+		ksDel (alreadyUsedMountpoints);
+		return false;
+	}
+
+	/* STEP 2: Check for wrong namespace (proc) */
+	if (keyGetNamespace (mountPoint) == KEY_NS_PROC)
+	{
+		// TODO: set error key (was exception in c++ code)
+		fprintf (stderr, "proc:/ mountpoint not allowed!\n");
+		ksDel (backendInfos);
+		ksDel (alreadyUsedMountpoints);
+		return false;
+	}
+
+	/* STEP 3: Check for name match */
+	const char * mpName = keyName (mountPoint);
+	if (keyGetNameSize (mountPoint) == 2 && *mpName == '/')
```

Comment:
Seems rather complicated... Why not just do a cascading lookup and report the found key in the error?

```c
Key * found = ksLookupByName (alreadyUsedMountpoints, mpName, 0);
if (found != NULL) {
    fprintf (stderr, "Mountpoint '%s' not possible, because mountpoint '%s' already exists.\n", mpName, keyName (found));
}
```

This should work for all cases of `mpName`